ste/sourceInfoProvider.go

```diff
@@ -71,6 +71,12 @@ type IBlobSourceInfoProvider interface {
 	BlobType() azblob.BlobType
 }
 
+type ISDDLBearingSourceInfoProvider interface {
+	ISourceInfoProvider
+
+	GetSDDL() (string, error)
```

Comment:
I have a hard time agreeing with either of these suggestions, naming wise. ISDDLBearingSourceInfoProvider is implemented by both fileSourceInfoProvider (on all OSes) and localFileSourceInfoProvider (on windows).

I feel like GetRawSDDL is... wrong in the case of Windows->Files, since we aren't _getting_ the raw SDDL, we're getting the portable one.

I feel like GetPortableSDDL is _maybe_ the right option, but at the same time, it kind of implies we're actively altering the SDDL we obtain (which, from windows, we are, but from files, it's already portable, so we aren't doing anything to it.)

Perhaps both may be an option? The raw option probably would never end up getting used, but it'd be nice to have.

WRT always returning an SDDLString-- This isn't necessary on files and could actually cause problems when we try to stringify, because the TranslateSID function doesn't know if the SID was already translated or not. Furthermore, I suspect it may actually error out if the SID is outright not present locally (investigation _may_ be needed).

Reply:
OK. Let's leave it as-is.